client/asset/interface.go

```diff
@@ -10,6 +10,13 @@ import (
 	"decred.org/dcrdex/dex"
 )
 
+type WalletTrait uint64
+
+const (
+	WalletTraitRescanner WalletTrait = 1 << iota
+	WalletTraitSender                // a Send method to complement Withdraw
```

Comment:
Here's where we could also have a `Sweeper`'s trait. See: https://github.com/decred/dcrdex/pull/1355#discussion_r775917934

Reply:
I love this, but I wish it were only used at a higher level, e.g. as a field of `core.xcWallet` that is populated when the wallet is opened based on interface type-checks. Having the assets self-report sets up potential inconsistencies between the UI and the available API.

Reply:
Can do, I think.  It would be better for the implementations to set these traits based on their methods.  Will just have to work out how to get this info to frontends then.  Because I'm doing the following to communicate this based on the walletState: https://github.com/decred/dcrdex/pull/1374/files#diff-a0c3b3a03517588f82d5bb3553e3e46ef5377c8bbcb4f8813ebb6d3cb0a6ee86R128-R139

Reply:
Hmm, I suppose the answer would be to put it in the `WalletState`s rather than the `WalletDefinition`s.  I think I considered that and ended up thinking it was not really "state" but then again we have Type there too.

Reply:
`WalletState` is what I was thinking.

Reply:
The bitfield could be created in `asset.OpenWallet` too and added as a return argument.

Reply:
Moved it up to core (`xcWallet` and `WalletState`) in https://github.com/decred/dcrdex/pull/1375